keyboards/crkbd/keymaps/soundmonster/keymap.c

```diff
@@ -301,20 +301,23 @@ void suspend_power_down_user() {
 }
 
 bool oled_task_user(void) {
-    if (timer_elapsed32(oled_timer) > 30000) {
-        oled_off();
-        return false;
-    }
-#ifndef SPLIT_KEYBOARD
-    else { oled_on(); }
-#endif
-
+#if defined(SPLIT_KEYBOARD)
     if (is_keyboard_master()) {
-        render_status_main();  // Renders the current keyboard state (layer, lock, caps, scroll, etc)
+#endif
+        if (timer_elapsed32(oled_timer) > 30000) {
```

Comment:
So rather than handling this manually you could use ` if (last_input_activity_elapsed() > 30000)` in either case you likely want to disable the built in timeout with `#define OLED_TIMEOUT 0` in config.h.